src/test/java/edu/carroll/dogtag/service/LoginServiceImplTest.java

```diff
@@ -0,0 +1,554 @@
+package edu.carroll.dogtag.service;
+
+import edu.carroll.dogtag.jpa.model.Login;
+import edu.carroll.dogtag.web.form.LoginForm;
+import jakarta.transaction.Transactional;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.springframework.test.util.AssertionErrors.*;
+
+/**
+ * Unit test to pass happy, crappy and crazy inputs to the loginService, and
+ * it should handle all situation that are tested and return the expected results
+ * a user must be registered to be in the database so that a login can be attempted
+ * If no user is present in the database it will always come back with null for returning
+ * a Login object.
+ */
+@SpringBootTest
+@Transactional
+public class LoginServiceImplTest {
+    private static final String user = "testuser";
+    private static final String password = "testpass";
+
+    private static final String email = "dgardner@gmail";
+
+    @Autowired
+    private LoginService loginService;
+
+    @Autowired
+    private RegisterService registerService;
+
+    @Test
+    public void validateUserNoUsersInDatabase() {
+        final LoginForm form = new LoginForm(user, password);
+        assertFalse("validateUserNoUsersInDatabase: should fail with no users in database", loginService.validateUser(form));
+        assertNull("User should be found in the database {}", loginService.findLogin(user));
+    }
+
+    @Test
+    public void validateUserNoUsersInDatabase2() {
+        final LoginForm form = new LoginForm(password, user);
+        assertNull("User should be found in the database {}", loginService.findLogin(user));
+        assertFalse("validateUserNoUsersInDatabase: should fail with no users in database and switched loginform inputs {}",
+                loginService.validateUser(form));
+    }
+
+    @Test
+    public void validateUserSuccessTest() {
+        Login createUser = new Login();
+        createUser.setPassword(password);
+        createUser.setUser(user);
+        createUser.setEmail(email);
+        final LoginForm form = new LoginForm(user, password);
+        assertTrue("Register should return true {}", registerService.register(createUser));
+        assertNotNull("User should be found in the database {}", loginService.findLogin(user));
```

Comment:
The test is Ok, but you don't verify the registered user's information matches that is returned by findLogin.  You could have a bug where the email was not stored, and if so, this test would pass, but not work properly.